extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -72,16 +78,35 @@ class AmpYoutube extends AMP.BaseElement {
     iframe.width = this.width_;
     iframe.height = this.height_;
     this.element.appendChild(iframe);
-    /** @private {?Element} */
+
+    /** @private {!Element} */
     this.iframe_ = iframe;
 
-    // TODO(mkhatib, #2050): Use PlayerReady message for layout promise.
-    return loadPromise(iframe);
+    /** @private @const {!Promise} */
+    this.playerReadyPromise_ = this.getPlayerReadyPromise_();
+
+    this.getWin().addEventListener(
```

Comment:
@cramforce does the `addEventListener` need to happen on the global `window` object? I see in the `iframe-helper.js` we're adding the `message` listeners on `iframe.ownerDocument.defaultView`.

`this.getWin()` in the context of the tests (and probably any iframe'd AMP document) will not be referencing the global `window` object. This might cause troubles, no? Should I just use `window.addEventListener` here instead of `this.getWin()`?